android/src/main/java/com/bitmovin/player/reactnative/BitmovinBaseModule.kt

```diff
@@ -0,0 +1,50 @@
+package com.bitmovin.player.reactnative
+
+import com.facebook.react.bridge.*
+import com.facebook.react.uimanager.UIManagerModule
+
+abstract class BitmovinBaseModule(
+    protected val context: ReactApplicationContext,
+) : ReactContextBaseJavaModule(context) {
+    /** Run [block] in [UIManagerModule.addUIBlock], forwarding the result to the [promise]. */
+    protected inline fun <T> addUIBlock(promise: Promise, crossinline block: () -> T) {
+        val uiManager = runAndRejectOnException(promise) { uiManager() } ?: return
+        uiManager.addUIBlock {
+            runAndRejectOnException(promise) {
+                promise.resolve(block())
+            }
+        }
+    }
+
+    /**
+     * Helper function that returns the initialized [UIManagerModule] instance or throw.
+     */
+    protected fun uiManager(): UIManagerModule =
```

Comment:
Those methods were duplicated in each module. Moved them here to deduplicate. They should probably be properties but didn't have time to refactor that.

Reply:
As a follow up, those methods should be only accessible in the `runAndRejectOnException` as their nullable alternative should be used instead to be able to set `promise.reject`. They should also be properties.

Reply:
Done